### ISSUE TITLE
Checkbox time to mark in list

### ISSUE THREAD

**Issue by rjlinuxekoos** [adamghill/django-unicorn#250]
Hi! This project is really good! I have a problem where I have a form with a list of orders and each order with its check. When checking the check, I go to the python code to store that checked order. However, the user marks many orders very quickly and this makes them lose which ones are marked previously. Thanks.

**Comment by adamghill** [comment]
A few different options I can think of:
- add a `defer` modifier to each checkbox and add a submit button with an action so that the checkboxes get sent in one call
- you could try using the experimental queue which attempts to batch up fast updates: https://www.django-unicorn.com/docs/queue-requests/ -- when I've tested it I've used a shared cache like `redis` so multiple web servers use the same serial queue

**Comment by adamghill** [comment]
Two other options I thought about that might be helpful:

- if the checkboxes are bound as a `model`, then you can set the `debounce` modifier to 0 (however, if it's an `action` the `debounce` is already 0)
- you might be able to use either a [loading state](https://www.django-unicorn.com/docs/loading-states/) or [dirty state](https://www.django-unicorn.com/docs/dirty-states/) to disable the checkboxes when an update is happening to prevent a user from clicking them all super fast

**Comment by rjlinuxekoos** [comment]
Good Morning !
The thing is, each checkbox has a unicorn:click and also a unicorn:partial.id to render only a part of the screen, I noticed that if my table has many registers it takes time to check the check now if it has 3 registers for example fast mark, because then it goes there in the view and back quickly, would you have any idea for me to optimize this process, as I mentioned above, is it correct?

**Comment by rjlinuxekoos** [comment]
unicorn:click it goes into the view and does the tasks very fast, but it takes a long time to get back to the template, even when rendering partially.

**Comment by adamghill** [comment]
I assume you mean with something like `document.getElementById("some-unicorn-element-id").value = 'some new value'`? Unfortunately, I don't think setting a value like that triggers a `change` event which is what `Unicorn` listens for. It doesn't look like it triggers any event...

Maybe there could be a method on the JavaScript `Unicorn` object to trigger an event to fire? Something like `Unicorn.trigger('hello-world-component', 'some-unicorn-element-id')`? Does something like that even make sense?

**Comment by rjlinuxekoos** [comment]
Yes, it would make a lot of sense if there was something like that!

Because we were able to solve the checkbox issue with vue.js but to return it to unicorn view was a little tricky, I only managed it through unicorn:model.lazy that listens to blur change, now with unicorn:model.defer I couldn't .

**Comment by adamghill** [comment]
I added `Unicorn.trigger` in 2d48fca which just got merged into master, so it will be included in the next release.

**Comment by rjlinuxekoos** [comment]
Hello good afternoon!

Could you give us a practical example of how we are going to use this new implementation?

Thank you very much in advance!

**Comment by adamghill** [comment]
Sure! The first argument is the component name and the second argument is the key. An example of how I've been testing would be:

```hello-world.html
<input unicorn:model="name" id="nameId" unicorn:key="nameKey" value="initial value" />

<script>
  document.getElementById("nameId").value = "new value";
  Unicorn.trigger("hello_world", "nameKey");
</script>
```

**Comment by rjlinuxekoos** [comment]
Understand! It will be very useful, thank you very much!

**Comment by rjlinuxekoos** [comment]
A suggestion that we had also thought of is the following: there in views, declare an array, and link this array with the checks of the screen, then according to what was marked, it would feed the array. Just a suggestion...

**Comment by adamghill** [comment]
The `Unicorn.trigger()` method is now released in 0.35.0. Documentation is at https://www.django-unicorn.com/docs/advanced/#trigger-model-update. I'm going to close this issue, but let me know if you run into any other problems. 👍
